donate help v4

Sritterbush opened the pull request:
Added the caller reference, which was missing from the previous pull review.

Sritterbush committed:
Update social.py

Updated Donate helpfile
'Costs 5 AP' changed to 'Costs 1 AP'

Commit:
Automate Helpfile

Alters the donate helpfile so that it will automatically update the AP value the helpfile lists in the future, should the cost to run the command ever change again.

A slightly more complex test run to be sure I'm doing this right.

Commit:
Update social.py

Commit:
Revert "Update social.py"

This reverts commit 1f29713b60fa3c4755dd2b938fc2185d05ac6b53.

Commit:
Revert "Automate Helpfile"

This reverts commit a10a5f41440812781f31300a0f9cea4c828fcf30.

Sritterbush committed:
Merge pull request #2 from Arx-Game/stable_orphan

Update social.py (#240)

Commit:
Update social.py

Commit:
Merge branch 'stable_orphan' of https://github.com/Sritterbush/arxcode into stable_orphan

Commit:
Update social.py

Commit:
Update home.py

TehomCD commented on commands/base_commands/social.py:
```diff
@@ -2618,13 +2618,19 @@ class CmdDonate(ArxCommand):
     action_point_cost = 1
 
+    def get_help(self, caller, cmdset):
+        caller = caller.player_ob
```
You don't need this line here, since `caller` isn't used for the rest of the function. Unused local variables fall into the category of 'linting errors' or warnings: things which won't cause any errors in code, but are things you should avoid. A lot of code editors will flag them to let you know to fix them.

Sritterbush replied:
Ah, I see! I thought it would be alright without, but talking to someone else gave me the impression I should include it. My confusion entirely.

TehomCD commented on commands/cmdsets/home.py:
```diff
@@ -1121,7 +1121,7 @@ def pay_owner(self, price, msg):
         loc.db.shopowner.pay_money(-price)
         assets = loc.db.shopowner.player_ob.assets
-        if price >= assets.min_silver_for_inform:
+        if price - (price * (self.get_discount() / 100.0)) >= assets.min_silver_for_inform:
```
I believe the price here is already discounted. Did you mean `+` rather than `-`, so it'd use the undiscounted price for sending an inform? I'm not sure I follow what the intent is here. 

A best practice is often to make different commits and even separate pull requests for unrelated changes: it makes it easier to see what your intent is during code reviews and helps in version control to identify when things changed and how.

Sritterbush replied:
I didn't realize this pushed through. I was testing an edit based on a bug request, didn't think it would append to my existing pull request.

TehomCD replied:
Ah! Okay, so the deal is that every pull request is tied to a branch: when you make a pull request, you're comparing the set of changes of commits on one branch to a branch you want to merge them into. Once the PR is opened, any further commits on that branch are added to it. To make a separate pull request, you'd make a different branch with the new set of changes.

Sritterbush replied:
So while I have a pending pull request in, all the updates to my own git automatically alter my pull request in process. I didn't know that before, but good info. I'll be mindful of this moving forward.

TehomCD replied:
Well, on that branch: you can have multiple branches coexisting simultaneously and switch between them. It'd be like:
```
git checkout stable_orphan
git pull upstream
git checkout -b fix-something
git commit -am "Fixed thing!!"
git push -u origin fix-something
git checkout stable_orphan
git checkout -b fix-other-thing
git commit -am "Fixed other thing!"
git push -u origin fix-other-thing
```
That'd get you two independent branches with different histories/changes.

Sritterbush replied:
I'm not seeing that flow of commands, since I'm using Github Desktop. I have forsaken my command prompt origins.

TehomCD commented:
Oh one thing that is helpful, if you don't mind, is to link to issues that you're addressing in a PR in the description or the commits. If you write, say, `Closes #236 ` in a commit message, it'll automatically close issue 236 if the PR is merged. In github, writing the pound sign followed by an issue number will create a link in that issue back to where it's referenced.

Thank you for your work in making PRs! These are looking very good for someone so new to it.

Commit:
Update social.py

Sritterbush commented:
Included an adjustment to the price code that should resolve  the issue in issue 236 where a shop asks you to have the silver on hand needed to make the purchase, even when you have a discount. This check appears to not factor in the discount because it happens before the buy command is validated.

Closes #236

Commit:
Update home.py

Sritterbush commented:
Reviewed the edit after your comment, realized I hit the code in the wrong place. Moved the edit to what I now believe is the correct price reference, in line 1199. I am... Very new to this. Not used to trying to test code without running it.

Sritterbush commented on commands/cmdsets/home.py:
```diff
@@ -1151,7 +1151,7 @@ def check_blacklist(self):
             return True
         for org in caller.player_ob.Dominion.current_orgs:
-            if org.name in blacklist:
+            if org.name in blacklist and caller in org.name = [ob.player.player for ob in org.active_members if not ob.secret]:
```
Okay, so I had a /third/ edit in process I didn't remember I was looking at. FML. This was in response to issue #231, but I wasn't sure it was a valid solution yet. I don't quite understand the code behind it.

Sritterbush replied:
Reverted this edit, it failed the Travis CI

TehomCD reviewed (commented):
You're doing just fine. :) I have to go to bed, so I'll review stuff tomorrow. Thanks again for your work!

Commit:
Update home.py

Reverted the change due to the Travis CI error. Not a valid way to write this solution.

TehomCD commented:
Looks good to me. Thanks! Remember to pull changes after this is merged to update your local repository.

TehomCD commented:
The easiest thing to do would be go to your fork on GitHub, hit the button to sync it up, then pull the changes locally. Then branches from that point will be up to date.
